components/eventing-controller/controllers/subscription/nats/integration_test.go

```diff
@@ -0,0 +1,837 @@
+		},
+
+		{
+			name: "invalid sink; not a valid cluster local service",
+			subscriptionOpts: []reconcilertesting.SubscriptionOpt{
+				reconcilertesting.WithFilter(reconcilertesting.EventSource, reconcilertesting.OrderCreatedEventTypeNotClean),
+				reconcilertesting.WithWebhookForNATS(),
+				reconcilertesting.WithSinkURL(
+					reconcilertesting.ValidSinkURL(ens.subscriberSvc.Namespace, "testapp")),
+			},
+			expect: expect{
+				k8sSubscription: []gomegatypes.GomegaMatcher{
+					reconcilertesting.HaveCondition(
+						conditionInvalidSink("sink is not valid cluster local svc, failed with error: Service \"testapp\" not found")),
+				},
+				k8sEvents: []v1.Event{
+					eventInvalidSink("Sink does not correspond to a valid cluster local svc")},
+			},
+		},
+
+		{
+			name: "valid sink",
+			subscriptionOpts: []reconcilertesting.SubscriptionOpt{
+				reconcilertesting.WithFilter("", reconcilertesting.OrderCreatedEventTypeNotClean),
+				reconcilertesting.WithSinkURL(validSinkURL(ens)),
+			},
+			expect: expect{
+				k8sSubscription: []gomegatypes.GomegaMatcher{
+					reconcilertesting.HaveCondition(reconcilertesting.DefaultCondition("")),
+					reconcilertesting.HaveSubscriptionReady(),
+				},
+			},
+			shouldTestDeletion: true,
+		},
+
+		{
+			name: "valid sink; with port",
+			subscriptionOpts: []reconcilertesting.SubscriptionOpt{
+				reconcilertesting.WithFilter("", reconcilertesting.OrderCreatedEventTypeNotClean),
+				reconcilertesting.WithSinkURL(validSinkURL(ens, ":8080")),
+			},
+			expect: expect{
+				k8sSubscription: []gomegatypes.GomegaMatcher{
+					reconcilertesting.HaveCondition(reconcilertesting.DefaultCondition("")),
+					reconcilertesting.HaveSubscriptionReady(),
+				},
+			},
+			shouldTestDeletion: true,
+		},
+
+		{
+			name: "valid sink; with port and endpoint",
+			subscriptionOpts: []reconcilertesting.SubscriptionOpt{
+				reconcilertesting.WithFilter("", reconcilertesting.OrderCreatedEventTypeNotClean),
+				reconcilertesting.WithSinkURL(validSinkURL(ens, ":8080", "/myEndpoint")),
+			},
+			expect: expect{
+				k8sSubscription: []gomegatypes.GomegaMatcher{
+					reconcilertesting.HaveCondition(reconcilertesting.DefaultCondition("")),
+					reconcilertesting.HaveSubscriptionReady(),
+				},
+			},
+			shouldTestDeletion: true,
+		},
+
+		{
+			name: "empty protocol, protocol setting and dialect",
+			subscriptionOpts: []reconcilertesting.SubscriptionOpt{
+				reconcilertesting.WithFilter("", reconcilertesting.OrderCreatedEventTypeNotClean),
+				reconcilertesting.WithSinkURLFromSvc(ens.subscriberSvc),
+			},
+			expect: expect{
+				k8sSubscription: []gomegatypes.GomegaMatcher{
+					reconcilertesting.HaveCondition(reconcilertesting.DefaultCondition("")),
+				},
+				natsSubscription: []gomegatypes.GomegaMatcher{
+					reconcilertesting.BeNotNil(),
+					reconcilertesting.BeValid(),
+					reconcilertesting.HaveSubject(reconcilertesting.OrderCreatedEventType),
+				},
+			},
+			shouldTestDeletion: true,
```

Comment:
Why do we sometimes test the deletion and sometimes not ? 🤔 Can't we just always test that the deletion works ?

Reply:
I am also not sure. I carried this over from the original tests. My hunch was, that the deletion originally only was done to  prevent collisions with subscription names due to the old tests setup. But since i can't say for sure, I kept it. Further, my arg for not doing it always, that this restructure was supposed to be a step away from test always everything. I was also half-in-favor for always deleting, but came decided to go the current way.

Reply:
forgot to add here, that after a conversation with the po, i made deletion tests obligatory